src/OrchardCore/OrchardCore.Abstractions/Localization/CultureScope.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using System.Globalization;
+
+namespace OrchardCore.Localization
+{
+    public sealed class CultureScope : IDisposable
+    {
+        public CultureScope(CultureInfo culture)
+        {
+            Culture = culture;
+            CultureInfo.CurrentUICulture = culture;
```

Comment:
What is the rule to either update CurrentCulture, CurrentUICulture, or both?
When would it make sense to not update the two with the same value?

Reply:
Just for info, maybe useful ;)

https://github.com/joakimriedel/culture-switcher/blob/master/src/CultureSwitcher.cs

Reply:
CurrentUICulture worked fine but why not having both. Felt like this was related to UI only on my PR. Now, the helper is kind of over-engineering like seb says but I'm fine with it if done right.